### PR TITLE
Savaradh containerregistry microsoft.container registry 2024 11 01 preview

### PR DESCRIPTION
**Pull request by savaradh**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/containerregistry/resource-manager/readme.md [hunk 1]
```diff
@@ -26,10 +26,18 @@ These are the global settings for the ContainerRegistry API.
...
+tag: package-2024-11-preview
 ```
 
 
+### Tag: package-2024-11-preview
+
+These settings apply only when `--tag=package-2024-11-preview` is specified on the command line.
+
+```yaml $(tag) == 'package-2024-11-preview'
+input-file:
```
Comment (razvanbadea-msft) [inline specification/containerregistry/resource-manager/readme.md:38]: missing input file

**Comment by razvanbadea-msft** [comment]
The first commit needs to be an exact copy of the previous API version. All new changes should only be added in the subsequent commits. 
This allows the reviewer to get a clear understanding of the actual changes being introduced. With the way the PR is raised now, it is not possible for the reviewer to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation. If you are doing the later option please indicate which commit is the exact copy of the previous version.

**Comment by savaradh** [comment]
@razvanbadea-msft I was following the new [API guidelines](https://eng.ms/docs/products/azure-developer-experience/design/api-specs/api-specs?tabs=openapi) and the first three commits were added by the script `Copy-ApiVersion.ps1`. I noticed that the script messed up the new API directory structure and the readme file. Had to fix those in the next two commits.
Sixth [commit](https://github.com/Azure/azure-rest-api-specs/pull/31542/commits/eefd30bdf3a8a6e59f3cc6fa5abe304fd617c4e0) is the one that shows all the new properties and changes. Hope it is clear and as expected. Please let me know if you still want me to abandon the PR and recreate a new one. Honestly, the script does a really bad job. IMO, simple copy/paste and a search/replace is a much better option.

**Comment by savaradh** [comment]
Raised an issue for python sdk failue - https://github.com/Azure/autorest.python/issues/2924
All the other SDKs work as expected

**Comment by savaradh** [comment]
Closing this pull request as a new PR is created as requested - https://github.com/Azure/azure-rest-api-specs/pull/31612

**Pull request closed without merging**